bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLoggerAllocator.java

```diff
@@ -127,10 +136,16 @@ BufferedLogChannel createNewLog(File dirForNextEntryLog) throws IOException {
 
     BufferedLogChannel createNewLogForCompaction(File dirForNextEntryLog) throws IOException {
         synchronized (createCompactionLogLock) {
-            return allocateNewLog(dirForNextEntryLog, COMPACTING_SUFFIX);
+            BufferedLogChannel bc = allocateNewLog(dirForNextEntryLog, COMPACTING_SUFFIX);
+            writingCompactingLogId = bc.getLogId();
+            return bc;
         }
     }
 
+    void clearCompactingLogId() {
+        writingCompactingLogId = -1;
```

Comment:
Do we need to guard this by `synchronized (createCompactionLogLock)`?  It seems not possible to have race conditions, but I see the `createNewLogForCompaction(File dirForNextEntryLog)` is guarded.

Reply:
No need this, the `createNewLogForCompaction` and `clearCompactingLogId` are serial.

Reply:
> No need this, the `createNewLogForCompaction` and `clearCompactingLogId` are serial.

What do you mean?

Reply:
Just wondering if there's a chance that writingLogId and writingCompactingLogId field changes wouldn't be visible for another thread that reads these fields.

Reply:
> Just wondering if there's a chance that writingLogId and writingCompactingLogId field changes wouldn't be visible for another thread that reads these fields.

Make sense, make it volatile.